src/lib/provable/merkle-list.ts

```diff
@@ -506,6 +530,200 @@ class MerkleListIterator<T> implements MerkleListIteratorBase<T> {
   }
 }
 
+class MerkleListIteratorFromEnd<T> implements MerkleListIteratorBase<T> {
```

Comment:
I expected this to be doable as part of the same iterator class 🤔

Reply:
yep - done, I wanted to maintain the same API without introducing a new equivalent for `next`, such as `nextReverse`, I don't like the workaround  but lmk what you think, maybe this can also be improved